clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -483,8 +487,19 @@ invariant_checker_t::parallel_shard_memref(void *shard_data, const memref_t &mem
                         "Kernel event marker value missing");
         if (memref.marker.marker_type == TRACE_MARKER_TYPE_KERNEL_EVENT &&
             // Give up on back-to-back signals.
```

Comment:
@derekbruening I assume back-to-back means

```
kernel xfer
syscall xfer
kernel xfer
...
```

Why do we "give up" on this?

Reply:
I assume b/c it's complex.  But isn't this a case this PR is trying to solve?  So the code is already trying to avoid raising an error, but fails?

Reply:
No this PR is solving nested signals without an intervening instr.

Reply:
The added tests show the exact case.